File: hw/codena/u3/hw22MazeDiagramClass/Code/Java/MazeGenerator/src/main/java/ec/edu/espe/mazegenerator/model/Maze.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.mazegenerator.model;

/**
 *
 * @author Daniel Codena, CodeBreakers, @ESPE
 */
public class Maze {

    private int rows;
    private int columns;
    private Room[][] rooms;
    private Path path;

    public Maze() {
    }

    public Maze(int rows, int columns, Room[][] rooms, Path path) {
        this.rows = rows;
        this.columns = columns;
        this.rooms = rooms;
        this.path = path;
    }

    public int getRows() {
        return rows;
    }

    public void setRows(int rows) {
        this.rows = rows;
    }

    public int getColumns() {
        return columns;
    }

    public void setColumns(int columns) {
        this.columns = columns;
    }

    public Room[][] getRooms() {
        return rooms;
    }

    public void setRooms(Room[][] rooms) {
        this.rooms = rooms;
    }

    public Path getPath() {
        return path;
    }

    public void setPath(Path path) {
        this.path = path;
    }

}

File: hw/sanchez/u3/hw22MazeSolidPrinciples/MazeApplicationPython/model/maze.py
from model.path import Path


class Maze:

    def __init__(self, n=0, m=0):
        self._n = n
        self._m = m
        self._rooms = [[None for _ in range(m)] for _ in range(n)]
        self._path = Path()

    @property
    def n(self):
        return self._n

    @property
    def m(self):
        return self._m

    @property
    def rooms(self):
        return self._rooms

    @property
    def path(self):
        return self._path

    def set_room(self, row, column, room):
        self._rooms[row][column] = room

    def get_room(self, row, column):
        return self._rooms[row][column]

    def __str__(self):
        return f"Maze({self._n} x {self._m})"